src/core/configuration/comparer.ts

```diff
@@ -4,12 +4,14 @@ import { mergeNameParts } from "./utils";
 
 interface IConfigChanges {
     options: Record<string, any>;
+    removedOptions: string[];
     templates: Record<string, ITemplate>;
 }
 
 function getChanges(current: IConfigNode, prev: IConfigNode) {
     const changesAccum: IConfigChanges = {
         options: {},
+        removedOptions: [],
```

Comment:
I suggest to add `appendRemovedValues` to the `changesAccum` object since the function is only called with `changesAccum.removedOptions` as 4th arg

smth like

```
function getChanges(current: IConfigNode, prev: IConfigNode) {
    const changesAccum: IConfigChanges = {
        options: {},
        removedOptions: [],
        templates: {},
        addRemovedValues(current: Record<string, any>, prev: Record<string, any>, path: string ) {
          appendRemovedValues(current, prev, path, this.removedOptions)
        }
    };

    compare(current, prev, changesAccum);

    return changesAccum;
}
```

Reply:
*and add corresponding  changes to the `IConfigChanges` interface